### DISCUSSION TITLE
How to get scattered photon information by beamnrc_gui?

### DISCUSSION THREAD

**Discussion by CQUswl** [nrc-cnrc/EGSnrc#930] [Questions]
I want to get information about scattered photons by using beamnrc, is it possible to get it by setting bits 24 to 28 of LATCH? How exactly can I achieve this?

**Comment by rtownson** [comment]
See page 113 in the BEAMnrc manual. 

Bits 24-28 store the region number where the secondary particle was created (if it is one). If 24-28 are all zero, then it's a primary particle. Bits 1-23 store information about where the particle was created/interacted (depending on your settings), if you have configured your CM regions to correspond to those bits.

https://nrc-cnrc.github.io/EGSnrc/doc/pirs509a-beamnrc.pdf

Edit: After the simulation, use BEAMdp to analyze the phase-space and filter according to the latch bits.

**Reply by CQUswl** [reply]
> See page 113 in the BEAMnrc manual.
> 
> Bits 24-28 store the region number where the secondary particle was created (if it is one). If 24-28 are all zero, then it's a primary particle. Bits 1-23 store information about where the particle was created/interacted (depending on your settings), if you have configured your CM regions to correspond to those bits.
> 
> https://nrc-cnrc.github.io/EGSnrc/doc/pirs509a-beamnrc.pdf
> 
> Edit: After the simulation, use BEAMdp to analyze the phase-space and filter according to the latch bits.

Thank you very much.I have some problem about the BEAMDP，what is the difference about the LATCH_filter_option_2 and the LATCH_filter_option_4？And are the Compton scattering photons secondary particle？What are the interactions？Exclude scattering particles with bits 1-23 or 24-28?

**Reply by rtownson** [reply]
I'm not sure what you're asking about beamdp - do you mean the inclusive v.s. exclusive latch filtering? I can't answer that for you, it depends on what you are looking at. The filtering allows you to plot quantities only of a subset of the particles, depending on their latch. So if you want to look at particles that interacted in the region corresponding to bit 1, you would include that bit and exclude the rest. Click the "?" button for more.

The example in the BEAMnrc manual answers the other questions you're asking. All discrete interactions are included. You probably want to use LATCH_OPTION=3, to set by interaction. See table 2 in the example. Note there is no way to record which type of interaction occurred.

**Reply by CQUswl** [reply]
> I'm not sure what you're asking about beamdp - do you mean the inclusive v.s. exclusive latch filtering? I can't answer that for you, it depends on what you are looking at. The filtering allows you to plot quantities only of a subset of the particles, depending on their latch. So if you want to look at particles that interacted in the region corresponding to bit 1, you would include that bit and exclude the rest. Click the "?" button for more.
> 
> The example in the BEAMnrc manual answers the other questions you're asking. All discrete interactions are included. You probably want to use LATCH_OPTION=3, to set by interaction. See table 2 in the example. Note there is no way to record which type of interaction occurred.

Why do I get different results when I exclude the same bit with the two options shown below in beamdp?
![图片](https://user-images.githubusercontent.com/114200268/197099797-489c6498-31af-4eef-856d-7d6a94fb1335.png)
![图片](https://user-images.githubusercontent.com/114200268/197099829-4d5173c1-c2f5-4681-862e-6dd33a21cabb.png)
So I would like to know what is the difference between these two options.I used LATCH_OPTION=3 in the phase space file.Is the region where photons interact different from the region where secondary photons are generated?

**Reply by rtownson** [reply]
The first option lets you exclude particles that have those bits flagged, which means they interacted (latch_option=3) in the corresponding regions. 

The second option checks bits 24-28 to determine where the particle was created. It then checks to see if that region corresponds to any of the regions setting bits 0-23 that you have checked.

So yes, the region where a photon interacts is different than where it was created, since a photon may interact several times along its path without being destroyed. Note that latches are inherited for latch_option=3, so a secondary particle will have the bits flagged for any of the regions where its primary interacted.

**Reply by CQUswl** [reply]
> The first option lets you exclude particles that have those bits flagged, which means they interacted (latch_option=3) in the corresponding regions.
> 
> The second option checks bits 24-28 to determine where the particle was created. It then checks to see if that region corresponds to any of the regions setting bits 0-23 that you have checked.
> 
> So yes, the region where a photon interacts is different than where it was created, since a photon may interact several times along its path without being destroyed. Note that latches are inherited for latch_option=3, so a secondary particle will have the bits flagged for any of the regions where its primary interacted.

Thanks，I understand！
